r1cs/gadgets/std/src/fields/nonnative/params.rs

```diff
@@ -175,3 +218,50 @@ pub(crate) const fn find_parameters(target_field_prime_length: usize, base_field
 
     (min_cost_num_of_limbs, min_cost_limb_size, min_cost)
 }
+
+#[cfg(test)]
+pub(crate) fn find_parameters<SimulationF: PrimeField, ConstraintF: PrimeField>(
+) -> (usize, usize, usize) {
+    return if let Some(c) = SimulationF::Params::DIFFERENCE_WITH_HIGHER_POWER_OF_TWO {
+        find_parameters_for_pseudomersenne(
+            SimulationF::size_in_bits(),
+            ConstraintF::size_in_bits(),
+            c,
+        )
+    } else {
+        find_parameters_for_generic_field(SimulationF::size_in_bits(), ConstraintF::size_in_bits())
+    };
+}
+
+// This test is only useful to print the constraints gain with and without pseudo-mersenne optimization.
+#[cfg(test)]
+pub(crate) fn test_different_constraint<
+    SimulationF: PrimeField,
+    ConstraintF: PrimeField,
+    R: rand::RngCore,
+>(
+    _rng: &mut R,
+) {
+    let (num_limbs, bits_per_limb, constraints) = find_parameters_for_pseudomersenne(
+        SimulationF::size_in_bits(),
+        ConstraintF::size_in_bits(),
+        SimulationF::Params::DIFFERENCE_WITH_HIGHER_POWER_OF_TWO.unwrap(),
+    );
+
+    let (num_limbs_unopt, bits_per_limb_unopt, constraints_unopt) =
+        find_parameters_for_generic_field(SimulationF::size_in_bits(), ConstraintF::size_in_bits());
+
+    if constraints != constraints_unopt {
+        println!(
+            "optimized: num limbs={}, bits_per_limb={}, cost={}, h={}",
+            num_limbs,
+            bits_per_limb,
+            constraints,
+            num_limbs * bits_per_limb - SimulationF::size_in_bits()
+        );
+        println!(
+            "unoptimized: num limbs={}, bits_per_limb={}, cost={}",
+            num_limbs_unopt, bits_per_limb_unopt, constraints_unopt
+        );
+    }
```

Comment:
A side question: why don't we also only print out in the case that non-optimized and optimized yield the same result?

Reply:
You mean why do we print only if optimized and non-optimized yield different results? I did it in thsi way when I wrote the test since it was run for all fields, not only for pseudo-mersenne ones, and so printing results was meaningless. Later on I have to modify the tests code and now the test is run only for pseudo-mersenne fields, so maybe now I can remove this restriction and let the test always printing out results. Thanks for pointing it out

Reply:
Done